### ISSUE TITLE
Miller-Rabin implementation doesn't check for odd numbers

### ISSUE THREAD

**Issue by miguevmartin** [sympy/sympy#27145]
The Sympy implementation of the Miller-Rabin primality testing algorithm (from sympy.ntheory.primetest import mr) doesn’t check whether n is an odd number, which results in simple tests like mr(90,[a]) returning True: https://github.com/sympy/sympy/blob/d2be7bacd2604e98a642f74028e8f0d7d6084f78/sympy/ntheory/primetest.py#L147-L187

**Comment by smichr** [comment]
It doesn't say that it is intended for odd numbers only so `if n < 2:` could be changed to `if n < 2 or not n%2:`.

**Comment by miguevmartin** [comment]
Yes, of course, that'd fix the issue, and thanks for following up. Hereby the importance of proofing APIs, especially cryptographic APIs (or APIs used to support crypto APIs), from potential misuses --intentional or otherwise.

**Comment by vprayag2005** [comment]
why is this issue  closed, the corresponding PR to this issue is still in draft?

**Comment by miguevmartin** [comment]
Reopened for further discussion.

**Comment by vprayag2005** [comment]
We don’t need to check for odd numbers separately since they can be either prime or composite. A simple fix is to update the condition to if n < 2 or (n > 2 and n % 2 == 0), as the previous approach incorrectly included 2. Let me know if this works, and I’ll implement it

**Comment by miguevmartin** [comment]
Yes, that would fix the issue. Yet a simpler logic expression would be: if n < 2 or n % 2 == 0. Thank you.

**Comment by vprayag2005** [comment]
> Yes, that would fix the issue. Yet a simpler logic expression would be: if n < 2 or n % 2 == 0. Thank you.

In this solution, 2 is also included, which is incorrect since 2 is a prime number.

**Comment by vprayag2005** [comment]
You can closed this issue as it is completed
